app/app.py

```diff
@@ -218,6 +221,61 @@ def initialize_schools(
                     )
 
 
+async def run_factories(db: AsyncSession, settings: Settings) -> None:
+    """Run the factories to create default data in the database"""
+    hyperion_error_logger = logging.getLogger("hyperion.error")
+    if not settings.USE_FACTORIES:
+        return
+
+    hyperion_error_logger.info("Startup: Factories enabled")
+    # Importing the core_factory at the beginning of the factories.
+    factories_list: list[Factory] = []
+    for module in all_modules:
+        if module.factory:
+            factories_list.append(module.factory)
+        else:
+            hyperion_error_logger.warning(
+                f"Module {module.root} does not declare a factory. It won't provide any base data.",
+            )
+
+    # We have to run the factories in a specific order to make sure the dependencies are met
+    # For that reason, we will run the first factory that has no dependencies, after that we remove it from the list of the dependencies from the other factories
+    # And we loop until there are no more factories to run and we use a boolean to avoid infinite loops with circular dependencies
+    action = True
```

Comment:
I'm not sure what name would be better, but I don't find `action` really explicit
```suggestion
    run_factory = True
```